mlir/lib/Bindings/Python/IRCore.cpp

```diff
@@ -1234,6 +1256,41 @@ PyOperationRef PyOperation::createInstance(PyMlirContextRef contextRef,
   return unownedOperation;
 }
 
+
+bool _nb_try_inc_ref(PyObject *obj) {
```

Comment:
We need to add the implementation for non-Py_GIL_DISABLED case:
https://github.com/wjakob/nanobind/blob/62fc996018d9ea4d51af9c86cf008c2562b4eeab/src/nb_ft.h#L16-L22